lib/course_planner/courses/offered_courses.ex

```diff
@@ -99,4 +99,19 @@ defmodule CoursePlanner.Courses.OfferedCourses do
          |> @notifier.notify_later()
        end)
   end
+
+  def load_offered_course_for_edit(id) do
+    offered_course =
+      OfferedCourse
+      |> Repo.get!(id)
+      |> Repo.preload([:term, :course, :students, :teachers])
+    changeset = OfferedCourse.changeset(offered_course)
+
+    {:ok, offered_course, changeset}
+  end
+
+  def can_teacher_edit_and_update_offered_course?(user, offered_course) do
```

Comment:
I would move this function to the module `CoursePlanner.Accounts.Teachers` and rename it to `can_update_offered_course?` so when you call it that will be `Teachers.can_update_offered_course?(user, offered_course)`, which seems better to me

Reply:
done